arch/lkl/kernel/setup.c

```diff
@@ -139,6 +141,10 @@ long lkl_sys_halt(void)
 	lkl_ops->sem_free(init_sem);
 
 	free_initial_syscall_thread();
+	if (lkl_netdevs_remove() == 0)
```

Comment:
I think we need some sort of cleanup infrastructure, where host can register cleanup routines. This would avoid propagation of specific host functions in the kernel code which IMO would be cleaner.

However, I think we can merge this PR as it is and I will log an issue to track that separately.